src/main/kotlin/brs/api/http/GetState.kt

```diff
@@ -25,18 +27,20 @@ internal class GetState(private val dp: DependencyProvider) :
         response.addProperty("lastBlock", dp.blockchainService.lastBlock.stringId)
         response.addProperty("cumulativeDifficulty", dp.blockchainService.lastBlock.cumulativeDifficulty.toString())
 
-        if (!"false".equals(request["includeCounts"], ignoreCase = true)) {
-            var totalEffectiveBalance: Long = 0
-            for (account in dp.accountService.getAllAccounts(0, -1)) {
-                val effectiveBalanceBURST = account.balancePlanck
-                if (effectiveBalanceBURST > 0) {
-                    totalEffectiveBalance += effectiveBalanceBURST
+        if (!"false".equals(request[INCLUDE_COUNTS_PARAMETER], ignoreCase = true)) {
+            if ("true".equals(request[INCLUDE_EFFECTIVE_BALANCE_PARAMETER], ignoreCase = true)) {
+                var totalEffectiveBalance: Long = 0
+                for (account in dp.accountService.getAllAccounts(0, -1)) {
```

Comment:
This loop on all accounts of the database take a long time and can be even seen as an attack vector. Now it is only executed if explicitly asked, maybe this should be even removed.